Tools/clinic/libclinic/utils.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+import dataclasses as dc
+from typing import (
+    Literal,
+    NoReturn,
+    overload,
+    TYPE_CHECKING,
+)
+
+if TYPE_CHECKING:
+    from clinic import Clinic
+
+
+# Clinic instance
+clinic: 'Clinic' | None = None
```

Comment:
This PR feels like it _only_ does my least favourite thing about your other PR :// https://github.com/python/cpython/issues/113299#issuecomment-1864092370